components/camel-azure/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/EventHubsConsumer.java

```diff
@@ -173,7 +173,12 @@ private void processCommit(final Exchange exchange, final EventContext eventCont
         try {
             var completionCondition = processCheckpoint(exchange);
             if (completionCondition.equals(COMPLETED_BY_SIZE)) {
-                eventContext.updateCheckpoint();
+                eventContext.updateCheckpointAsync()
+                        .subscribe(unused -> LOG.debug("Processed one event..."), error -> {
+                            LOG.debug("Error when updating Checkpoint: {}", error.getMessage());
+                            exchange.setException(error);
```

Comment:
Since the callback is async then in case of an error, then storing that on the exchange does not make sense as the exchange would likely no longer be in use. Instead the error should be logged, or better yet processed by exception handler on the consumer - there should be a getExceptionHandler.handleException(...)

Reply:
I decided to just log the error. Going deeper seems to be out of my experience level within this matter..